lib/appsignal/transaction.rb

```diff
@@ -520,17 +520,18 @@ def add_error(error, &block)
       return unless error
       return unless Appsignal.active?
 
-      _set_error(error) if @error_blocks.empty?
-
-      if !@error_blocks.include?(error) && @error_blocks.length >= ERRORS_LIMIT
-        Appsignal.internal_logger.warn "Appsignal::Transaction#add_error: Transaction has more " \
-          "than #{ERRORS_LIMIT} distinct errors. Only the first " \
-          "#{ERRORS_LIMIT} distinct errors will be reported."
+      if error.instance_variable_get(:@__appsignal_error_reported) && !@error_blocks.include?(error)
```

Comment:
I wonder if we should do something with the error block? Consider:

```ruby
Appsignal.report_error(error_wrapper) do
  Appsignal.add_tags(wrapper: true)
end

Appsignal.report_error(error_cause) do
  Appsignal.add_tags(cause: true)
end
```

If we're not reporting the error cause because we consider it to be equivalent to its wrapper, should we apply its corresponding error block to the wrapper?

Reply:
Yeah, I think it's likely not worth the effort in either scenario.